### PR TITLE
rgw/notification: enrich the expired notification log line for persistent notificatio

### PR DESCRIPTION
**Pull request by kchheda3**
To get better visibility on the notification entries that are expired post max_retry/timeout value, enrich the log line to publish the topic name, bucket name, object name and bucket owner along with max_retry_value and timeout/current time value

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/driver/rados/rgw_notify.cc [hunk 1]
```diff
@@ -201,8 +201,15 @@ class Manager : public DoutPrefixProvider {
     if ( (topic_persistency_ttl != 0 && event_entry.creation_time != ceph::coarse_real_clock::zero() &&
          time_now - event_entry.creation_time > std::chrono::seconds(topic_persistency_ttl))
          || ( topic_persistency_max_retries != 0 && entry_persistency_tracker.retires_num >  topic_persistency_max_retries) ) {
-      ldpp_dout(this, 20) << "Expiring entry retry_number=" << entry_persistency_tracker.retires_num << " creation_time="
-                          << event_entry.creation_time << " time_now:" << time_now << dendl;
+      ldpp_dout(this, 1) << "Expiring entry for topic= "
```
Comment (thotz) [inline src/rgw/driver/rados/rgw_notify.cc:204]: changing loglevel from 20 to 1 for this one may not right, a part from that change looks good

Comment (kchheda3) [reply src/rgw/driver/rados/rgw_notify.cc:204]: Its purposely changed to log level 1, so we can monitor the logs for messages that are expired.
unfortunately with log level 20, its super verbose and cannot be turned on all the time to monitor the expired notification.
Also expired notification is as good as errored notification, so to have some visibility need to make the log level 1.
this PR is inline with similar PR for sync notification (non-persistent notification) https://github.com/ceph/ceph/pull/52876/files

Comment (kchheda3) [reply src/rgw/driver/rados/rgw_notify.cc:204]: > changing loglevel from 20 to 1 for this one may not right, a part from that change looks good

Its purposely changed to log level 1, so we can monitor the logs for messages that are expired.
unfortunately with log level 20, its super verbose and cannot be turned on all the time to monitor the expired notification.
Also expired notification is as good as errored notification, so to have some visibility need to make the log level 1.
this PR is inline with similar PR for sync notification (non-persistent notification) https://github.com/ceph/ceph/pull/52876/files

Comment (yuvalif) [reply src/rgw/driver/rados/rgw_notify.cc:204]: Agree that this should be level 1.
Entry expiry may cause information loss, so shoudl be considered an error case.

**Comment by yuvalif** [comment]
jenkins test api

**Review by yuvalif — APPROVED** [review]

**Review by thotz — APPROVED** [review]

**Comment by yuvalif** [comment]
teuthology passing: http://pulpito.front.sepia.ceph.com/yuvalif-2023-09-11_12:35:49-rgw:notifications-wip-yuval-test-distro-default-smithi/

**Pull request merged**